services/explorer-indexer/src/modules/node/historicalSyncer.service.ts

```diff
@@ -163,114 +174,227 @@ export default class HistoricalSyncService {
+          this.txService.populateTransactionWithPBFT(tx, block)
         );
-
-        if (block.transactions[i]) {
-          const fetchStart = new Date();
-          const tx = await this.rpcConnector.getTransactionByHash(
-            block.transactions[i].hash
-          );
-          const fetchEnd = new Date();
-          console.log(
-            `Fetched tx ${block.transactions[i].hash} in ${
-              fetchEnd.getTime() - fetchStart.getTime()
-            }ms.`
-          );
-          blockTxs.push(
-            this.txService.populateTransactionWithPBFT(
-              this.txService.txRpcToITransaction(tx),
-              block
-            )
-          );
+        block.transactions?.forEach((tx: ITransaction) => {
+          const gasUsed = new BigInteger(tx.gasUsed.toString() || '0');
+          tx.gasUsed = gasUsed.toString();
+          const gasPrice = new BigInteger(tx.gasPrice.toString() || '0');
+          tx.gasPrice = gasPrice.toString();
+          const reward = gasUsed.multiply(gasPrice);
+          const newVal = blockReward.add(reward);
+          blockReward = newVal;
+        });
+
+        block.reward = blockReward.toString();
+        const blockToSave = { ...block };
+        blockToSave.transactions = block.transactions;
+        blockToSave.transactionCount = block.transactions?.length;
+
+        const savedBlock = await this.pbftService.safeSavePbft(blockToSave);
+
+        if (savedBlock) {
+          this.logger.log(`Finalized block ${savedBlock.hash}`);
         }
-      }
-      const fetchTxEnd = new Date();
-      console.log(
-        `Fetched ${block.transactions?.length} tx-es in ${
-          fetchTxEnd.getTime() - fetchTxStart.getTime()
-        } ms.`
-      );
-
-      if (!block.hash) {
-        continue;
-      }
 
-      let blockReward = new BigInteger('0', 10);
+        // update sync state
+        this.syncState.number = block.number;
+        this.syncState.genesis = this.chainState.genesis;
+        this.syncState.hash = block.hash;
 
-      // @note : Right now there is no way to get the genesis transactions to set:
-      // Initial validators, initial balances for delegators and faucet.
-      // big TODO
-
-      const txReceipts = await this.getTransactionReceipts(
-        blockTxs.map((tx) => tx.hash),
-        20
-      );
-
-      Array.from(new Set(blockTxs)).forEach((tx: ITransaction, idx: number) => {
-        const receipt = txReceipts[idx];
-        tx.status = parseInt(receipt.status, 16);
-        tx.index = parseInt(receipt.transactionIndex, 16);
-        const gasUsed = new BigInteger(receipt.gasUsed || '0x0', 16);
-        tx.gasUsed = gasUsed.toString();
-        const gasPrice = new BigInteger(tx.gasPrice || '0x0', 16);
-        tx.gasPrice = gasPrice.toString();
-        tx.cumulativeGasUsed = Number(
-          new BigInteger(receipt.cumulativeGasUsed || '0x0', 16)
+        const completed = new Date();
+        console.log(
```

Comment:
Let's use logger.debug as it will show in the console and we don't need to remove it later on